public/app/plugins/datasource/tempo/streaming.ts

```diff
@@ -100,6 +105,116 @@ export function doTempoChannelStream(
     );
 }
 
+export function doTempoMetricsStreaming(
+  query: TempoQuery,
+  ds: TempoDatasource,
+  options: DataQueryRequest<TempoQuery>
+): Observable<DataQueryResponse> {
+  const range = options.range;
+  const key = getLiveStreamKey();
+
+  let state: LoadingState = LoadingState.NotStarted;
+  const step = stepToNanos(query.step);
+
+  return getGrafanaLiveSrv()
+    .getStream<MutableDataFrame>({
+      scope: LiveChannelScope.DataSource,
+      namespace: ds.uid,
+      path: `metrics/${key}`,
+      data: {
+        ...query,
+        step,
+        timeRange: {
+          from: range.from.toISOString(),
+          to: range.to.toISOString(),
+        },
+      },
+    })
+    .pipe(
+      takeWhile((evt) => {
+        if ('message' in evt && evt?.message) {
+          const frameState: SearchStreamingState = evt.message.data.values[2][0];
+          if (frameState === SearchStreamingState.Done || frameState === SearchStreamingState.Error) {
+            return false;
+          }
+        }
+        return true;
+      }, true),
+      map((evt) => {
+        let newResult: DataQueryResponse = { data: [], state: LoadingState.NotStarted };
+        if ('message' in evt && evt?.message) {
+          // Schema should be [data, metrics, state, error]
```

Comment:
Do you think it's worth being future-proof and use schema provided in `evt.message.schema` to identify error field?